Resync css/css-scoping WPT

keithamus opened the pull request:
#### d4541a2f0f47536953f9e33e066a203fd2f138ba
<pre>
Resync css/css-scoping WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=281746">https://bugs.webkit.org/show_bug.cgi?id=281746</a>

Reviewed by Tim Nguyen.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/d00df073d0d5271402e526ab3cd846b2e4c63822">https://github.com/web-platform-tests/wpt/commit/d00df073d0d5271402e526ab3cd846b2e4c63822</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-001.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-002.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-003.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-004.tentative-expected.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-descendant-003-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-004.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-manual-assignment-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-manual-assignment.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-query-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-query-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-descendant-003-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-context-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-context-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-002-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-003-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-004-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-005-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-006-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-not-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/scope-pseudo-in-shadow-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/scope-pseudo-in-shadow.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/w3c-import.log:

Canonical link: <a href="https://commits.webkit.org/290130@main">https://commits.webkit.org/290130@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1ffd4e68b3ff171bd9da3e55d1ab195309a5d573

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89047 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8571 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43818 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94021 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39808 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8958 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16757 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68620 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26284 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92049 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6845 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80773 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48972 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6594 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/35163 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38915 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76956 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36130 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95858 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16227 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11868 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77490 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16483 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76575 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76778 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18928 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21169 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/19763 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9319 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16241 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15982 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19433 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17763 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-scoping/css-scoping-shadow-dynamic-remove-style-detached-expected.txt:
@mdubet is probably right, the WPE/GTK baselines for this test are not necessary.

nt1m replied:
(Also the common baseline is missing from this patch)

nt1m commented on LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-scoping/css-scoping-shadow-dynamic-remove-style-detached-expected.txt:
ditto

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/css-scoping-shadow-dynamic-remove-style-detached-expected.txt:
Why was this file removed? That's causing the failures for all the platforms. 

The only reason you don't see it on mac is because of: https://searchfox.org/wubkat/source/LayoutTests/platform/mac/TestExpectations#1868 , but we should be keeping the common baseline otherwise.

keithamus replied:
Oops yeah you're right. Added back.

nt1m commented on LayoutTests/TestExpectations:
```diff
@@ -5522,6 +5522,9 @@ webkit.org/b/214462 imported/w3c/web-platform-tests/css/css-scoping/host-context
 webkit.org/b/214462 imported/w3c/web-platform-tests/css/css-scoping/slotted-placeholder.html [ ImageOnlyFailure ]
 
+webkit.org/b/287169 imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button-expected.html [ ImageOnlyFailure ]
+webkit.org/b/287169 imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button-ref.html [ ImageOnlyFailure ]
```
You should be listing:

```
imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content.html [ ImageOnlyFailure ]
imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button.html [ ImageOnlyFailure ]
```
not the ref files.

keithamus replied:
I don't understand what happened here. I had these listed as such in the last push, but it seems to have regressed.

keithamus committed:
Resync css/css-scoping WPT
https://bugs.webkit.org/show_bug.cgi?id=281746

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/d00df073d0d5271402e526ab3cd846b2e4c63822

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-001.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-002.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-003.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-004.tentative-expected.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-descendant-003-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-flattened-004.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-manual-assignment-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-manual-assignment.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-query-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/has-slotted-query-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-descendant-003-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-context-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-context-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-in-host-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-002-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-003-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-004-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-005-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-multiple-006-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-not-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/scope-pseudo-in-shadow-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/scope-pseudo-in-shadow.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-details-content.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-file-selector-button.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/w3c-import.log:

Canonical link: https://commits.webkit.org/290130@main
